enhancements/cluster-logging/fine-grained-per-namespace-logs-access.md

```diff
@@ -0,0 +1,249 @@
+---
+title: fine-grained-logs-access
+
+authors:
+  - "@aminesnow"
+
+reviewers:
+  - "@periklis"
+  - "@shwetaap"
+  - "@alanconway"
+  - "@xperimental"
+  - "@jcantrill"
+
+approvers:
+  - "@periklis"
+
+creation-date: 2023-05-31
+last-updated: 2023-05-31
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/LOG-4020
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Openshift Logging: Fine grained access to logs
+
+## Summary
+
+This enhancement proposal seeks to outline the solution for a more fine grained access to LokiStack logs on OpenShift Container Platform 4, ie, have the ability as the cluster administrator to grant access to logs on a namespace basis.
+
+The proposal provides an overview of the changes required to achieve this goal as well as some implementation details.
+
+## Motivation
+
+In enterprise environments, where OpenShift is used across different legal entities, it's common to have central teams that support the application teams in the respective entities. But given that some application may log sensitive data, those centralized support teams are not granted access to logs but they can only view specific objects, such as pods in the namespace.
+
+Currently, access to logs in LokiStack is granted when a user has access to the given namespace or when the user is part of a specific cluster-admin Group. And even though OpenShift Container Platform 4 does allow to configure RBAC to address this issue, LokiStack does not, and therefore grants access to logs to people that should not see them.
+
+This enhancement proposal presents a solution that enables cluster admins to have a more fine grained control over who accesses what logs.
+
+### User Stories
+
+* Use Case number one, is where a user has access to a specific namespace to see the objects included but is denied to view logs from pods (for legal reason mostly).
```

Comment:
Consider writing this like:

As an administrator,
I want to restrict visibility of certain logs for users with limited access
because there are legal reasons to do so.

... or something like that:

As what role
What we are doing
and why